### ISSUE TITLE
Execute a query against a set of BSON objects

### ISSUE THREAD

**Issue by almightycouch** [Softmotions/ejdb#164]
Hello,

I'm trying to implement an embeddable MongoDB-like library using EJDB.
I was wondering why the [query language](http://ejdb.org/doc/ql/ql.html) implementation is not available through the public API.

I'm thinking about different scenarios where it would be very useful to execute a query against a set of BSON objects instead of a EJDB collection.

```
EJQ* query; // a query object

bson obj; // BSON object we'd like to match
if(ejdbqrymatch(query, &obj)) {
    printf("matched");
}
```

The `ejdbqrymatch()` function could look like the following snippet:

```
bool ejdbqrymatch(const EJQ *q, const bson *bs) {
    bool m = 1;
    const char *bsbuf = bson_data(bs);
    int bsz = bson_size(bs);
    int qsz = TCLISTNUM(q->qflist);
    for(int i = 0; i < qsz && m; ++i) {
        m = _qrybsmatch((EJQF*)(TCLISTVALPTR(q->qflist, i)), bsbuf, bsz);
    }
    return m;
}
```

If this seems interesting to you, i already have a few `.patch` files i'm using locally and could make available through a pull-request.

**Comment by GbalsaC** [comment]
@almightycouch refering to installation of https://github.com/almightycouch/meteorpp 
`cmake ..
make
`
Returns the following error:
`meteorpp/src/live_query.cpp:116:178: error: ‘ejdbqrymatch’ was not declared in this scope
         return ejdbqrymatch(std::shared_ptr<EJQ>(ejdbcreatequery2(_coll->_db.get(), bson_data(_selector.get())), ejdbquerydel).get(), collection::convert_to_bson(document).get());
                                                                                                                                                                                  ^
make[2]: *** [CMakeFiles/meteorpp.dir/src/live_query.cpp.o] Error 1
make[1]: *** [CMakeFiles/meteorpp.dir/all] Error 2
make: *** [all] Error 2
`
I assume this is the function refered to?

**Comment by adamansky** [comment]
Closed in favour to ejdb 2.0 https://github.com/Softmotions/ejdb/blob/master/README.md
